dev/com.ibm.ws.concurrent.persistent_fat/test-applications/schedtest/src/web/SchedulerFATServlet.java

```diff
@@ -2086,8 +2065,21 @@ public void testFindByName(PrintWriter out) throws Exception {
             throw new Exception("Should not be able to see tasks from outside of application. " + resultsByTaskId);
 
         // Wait for tasks C and D to complete their only execution.
-        pollForResult(statusC.getTaskId(), 1);
-        pollForResult(statusD.getTaskId(), 1);
+        for (long start = System.nanoTime(); System.nanoTime() - start < TIMEOUT_NS
```

Comment:
Instead of adding this logic inline, why not correct the logic in pollForResult?

Reply:
Because pollForResult was interpreted to mean different things to different tests that used it.  Its behavior was only incorrect if you are looking at one of the tests that didn't want it to behave the way it did, otherwise it was doing exactly the right thing, and altering it would break that.  It's a good example of why this sort of utility method is a bad practice in tests.  Developers writing a test think they know what it does based on its name and start to unknowingly misuse it.  The solution here is to remove it and let each test perform precisely the steps that it needs inline, which will be ideal for future debugging as well as for ensuring correctness.

Reply:
I understand that now, what you are doing makes sense to me.  Thank you for taking the time to explain that to me!